### DISCUSSION TITLE
Four filers, two shared sql store and three datacentr

### DISCUSSION THREAD

**Discussion by aoberest** [seaweedfs/seaweedfs#1949] [Q&A] [answered]
Hello.

I want to make a cluster in three datacenters. The master data will be located in two of them. I am planning two sql databases to store metadata in DC1 and DC2. Each database will serve filers in its DC. 

Look at the diagram, please.
![Schema1](https://user-images.githubusercontent.com/49243562/112945229-d400c500-9155-11eb-9d21-9e1ef3becf59.png)

All masters and volumes have connections. (peers)

How do I do replication between filers DC1 and filers DC2 - ?

Can I specify "peers" filer1DC1 <-> filer2DC1 < - > filer1DC2 < - > filer2DC2 - ?

How are there options for configuring filers - ?

Maybe there are other solutions ?

**Comment by chrislusf** [comment, marked as answer]
For cross data centers, you can setup 2 clusters, and use `weed filer.sync` to replicate synchronously. 

https://github.com/chrislusf/seaweedfs/wiki/Filer-Active-Active-cross-cluster-continuous-synchronization

**Reply by aoberest** [reply]
Thanks for the reply.

I'm trying to make a failover cluster. No SPOF.
I set the replication to 101.
I made the division into datacenters using the "-dataCenter "option, in order to achieve the replication goal of "replication 101"

If I use "Filer Active Active cross cluster continuous synchronization" then I can't get "No SPOF", right ?

Can I use the "[Filer Store Replication](https://github.com/chrislusf/seaweedfs/wiki/Filer-Store-Replication)" statement and set the peers option for each "Filer" ?

For example:
```weed filer -s3 -peers=filer1DC1:8888,filer2DC1:8888,filer1DC2:8888,filer2DC2:8888```

I tried this option and it seems to work, but I'm not sure if this is the correct setting for "filers"

```
weed version
version 30GB 2.31 f5b5e4d linux amd64
```

**Reply by chrislusf** [reply]
You can surely do so to achieve no SPOF at the cost of write latency, since cross-dc latency is higher and SeaweedFS needs strong consistency during writes.

**Reply by aoberest** [reply]
Thank you!
